Backend/Models/Users/UserManager.cs

```diff
@@ -40,12 +41,46 @@ public static bool IsUserAdmin(User user)
         {
             return user.Ip == "127.0.0.1";
         }
+
         public static User? GetUserByUsername(string username) 
         {
             return UsersList.Find(user => user.Username == username);
         }
-        public static void Authenticate(WebSocket socket, string username) { }
+
+        public static bool Authenticate(WebSocket socket, string username)
+        {
+            User? user = GetUserBySocket(socket);
+            if (user == null)
+            {
+                Console.WriteLine("Invalid authentication attempt");
+                return false;
+            }
+
+            if (user.IsRegistered)
+            {
+                Console.WriteLine($"Double registration prevented from: {user.Username}");
+                return false;
+            }
+
+            bool isUserConnected = GetUserByUsername(username) != null;
```

Comment:
hmm we should probably rename this variable to make it clear this is a username check, maybe isUsernameConnected?